### DISCUSSION TITLE
Buttons and JS

### DISCUSSION THREAD

**Discussion by IvyMasterDiego** [squidfunk/mkdocs-material#7067] [Q&A] [answered]
Is it possible to call a JS file from a button and how could I perform this.

For example:
```
[click-me](javascript/test.js){.md-button .md-button--primary}
```

Trying this I noticed that it is not working and is not giving me the expected result. I would like to know what is the correct way to implement this on my page.

**Comment by alexvoss** [comment, marked as answer]
The Markdown code you posted produces a link to the JavaScript. It  does nothing to execute it. The JavaScript in Material works the same as it would for any website. You can add extra JavaScript to be loaded into your pages with the [extra_javascript option](https://squidfunk.github.io/mkdocs-material/customization/?h=extra+javascript#additional-javascript). 

The best thing to do is to register a function that will run once the page is fully loaded and will itself register handlers for events on the buttons as needed. The basics of how to register such a callback are described in the "How to integrate with third-party JavaScript libraries" admonition in the documentation (as linked to above).

Hope this helps.

**Reply by IvyMasterDiego** [reply]
Can you show me an example because I tried to do it as it says on the page and it didn't work for me.

**Reply by alexvoss** [reply]
Here is an example project: 
[button-example.zip](https://github.com/squidfunk/mkdocs-material/files/15044301/button-example.zip)
